rpcs3/Emu/Cell/SPUAnalyser.cpp

```diff
@@ -12,9 +12,24 @@ std::shared_ptr<spu_function_t> SPUDatabase::find(const be_t<u32>* data, u64 key
 		const auto& func = found.first->second;
 
 		// Compare binary data explicitly (TODO: optimize)
-		if (LIKELY(func->size <= max_size) && std::memcmp(func->data.data(), data, func->size) == 0)
+		//if (LIKELY(func->size <= max_size) && std::memcmp(func->data.data(), data, func->size) == 0)
 		{
-			return func;
+			const auto dwords = (u32*)(func->data.data());
+			const auto cast_data = (u32*)(data);
+			const auto limit = std::min(max_size, func->size) >> 2;
+
+			bool failed = false;
+			for (u32 dword = 0; dword < limit; dword++)
```

Comment:
Are you sure this is better than simple `memcmp`? AFAIK it's usually vectorized as well.